tests/sentry/integrations/github/test_integration.py

```diff
@@ -738,3 +738,109 @@ def test_get_trees_for_org_rate_limit_401(self):
                 ("baz", "master", []),
             ]
         )
+
+    @responses.activate
+    def test_get_trees_for_org_makes_API_requests_before_MAX_CONNECTION_ERRORS_is_hit(self):
+        """
+        If some requests fail, but `MAX_CONNECTION_ERRORS` isn't hit, requests will continue
+        to be made to the API.
+        """
+        installation = self.get_installation_helper()
+        self.set_rate_limit()
+
+        # There's no response for repo `xyz` here because the one added in `setUp` is fine
+
+        # Given that below we mock MAX_CONNECTION_ERRORS to be 2, the error we hit here
+        # should NOT force the remaining repos to pull from the cache.
+        responses.replace(
+            responses.GET,
+            f"{self.base_url}/repos/Test-Organization/foo/git/trees/master?recursive=1",
+            body=ApiError("Server Error"),
+        )
+
+        # We should see the data from these responses, in spite of the error above.
+        responses.replace(
+            responses.GET,
+            f"{self.base_url}/repos/Test-Organization/bar/git/trees/main?recursive=1",
+            json={"tree": [{"path": "src/bar.py", "type": "blob"}]},
+            status=200,
+        )
+        responses.replace(
+            responses.GET,
+            f"{self.base_url}/repos/Test-Organization/baz/git/trees/master?recursive=1",
+            json={"tree": [{"path": "src/baz.py", "type": "blob"}]},
+            status=200,
+        )
+
+        # Clear the cache so we can tell when we're pulling from it rather than from an
+        # API call
+        cache.clear()
+
+        with patch(
+            "sentry.integrations.github.client.MAX_CONNECTION_ERRORS",
+            new=2,
+        ):
+
+            trees = installation.get_trees_for_org()
+            assert trees == self._expected_trees(
+                [
+                    ("xyz", "master", ["src/xyz.py"]),
+                    # foo isn't in the list because the request errors out.
+                    # bar and baz have the correct data, meaning their respective
+                    # API requests were in fact made.
+                    ("bar", "main", ["src/bar.py"]),
+                    ("baz", "master", ["src/baz.py"]),
+                ]
+            )
+
+    @responses.activate
+    def test_get_trees_for_org_falls_back_to_cache_once_MAX_CONNECTION_ERRORS_is_hit(self):
+        """Once `MAX_CONNECTION_ERRORS` requests fail, the rest will grab from the cache."""
+        installation = self.get_installation_helper()
+        self.set_rate_limit()
+
+        # There's no response for repo `xyz` here because the one added in `setUp` is fine
+
+        # Given that below we mock MAX_CONNECTION_ERRORS to be 1, the error we hit here
+        # should force the remaining repos to pull from the cache.
+        responses.replace(
+            responses.GET,
+            f"{self.base_url}/repos/Test-Organization/foo/git/trees/master?recursive=1",
```

Comment:
If make the error happen for `xyz` you won't need to make any changes to the responses for `bar` and `baz`.
You would not get the tree for any of the repos.

Just trying to minimize the number of lines per test to make it easier to read.

Reply:
Yeah, good idea. Will do.